apis/comms_api/scripts/wazuh_comms_apid.py

```diff
@@ -0,0 +1,209 @@
+
+
+def setup_logging(foreground_mode: bool) -> dict:
+    """Sets up the logging module and returns the configuration used.
+
+    Parameters
+    ----------
+    foreground_mode : bool
+        Whether to execute the script in foreground mode or not.
+
+    Returns
+    -------
+    dict
+        Logging configuration dictionary.
+    """
+    log_config_dict = set_logging(log_filepath=COMMS_API_LOG_PATH,
+                                  log_level='INFO',
+                                  foreground_mode=foreground_mode)
+    
+    for handler in log_config_dict['handlers'].values():
+        if 'filename' in handler:
+            utils.assign_wazuh_ownership(handler['filename'])
+            os.chmod(handler['filename'], 0o660)
+
+    logging.config.dictConfig(log_config_dict)
+
+    return log_config_dict
+
+def configure_ssl(keyfile: str, certfile: str) -> None:
+    """Generate SSL key file and self-siged certificate if they do not exist.
+
+    Raises
+    ------
+    ssl.SSLError
+        Invalid private key.
+    IOError
+        File permissions or path error.
+    """
+    try:
+        if not os.path.exists(keyfile) or not os.path.exists(certfile):
+            private_key = generate_private_key(keyfile)
+            logger.info(f"Generated private key file in {certfile}")
+            
+            generate_self_signed_certificate(private_key, certfile)
+            logger.info(f"Generated certificate file in {certfile}")
+    except ssl.SSLError as exc:
+        e = Exception('Private key does not match with the certificate')
+        raise e from exc
+    except IOError as exc:
+        if exc.errno == 22:
+            e = Exception('PEM phrase is not correct')
+        elif exc.errno == 13:
+            e = Exception('Ensure the certificates have the correct permissions')
+        else:
+            e = Exception('Wazuh comms API SSL error. Please, ensure the configuration is correct')
+        
+        raise e from exc
+
+def ssl_context(conf, default_ssl_context_factory) -> ssl.SSLContext:
+    """Returns the default SSL context with a custom minimum version.
+
+    Returns
+    -------
+    ssl.SSLContext
+        Server SSL context.
+    """
+    context = default_ssl_context_factory()
+    context.minimum_version = ssl.TLSVersion.MINIMUM_SUPPORTED
+    return context
+
+def get_gunicorn_options(pid: int, foreground_mode: bool, log_config_dict: dict) -> dict:
+    """Get the gunicorn app configuration options.
+
+    Parameters
+    ----------
+    pid : int
+        Main process ID.
+    foreground_mode : bool
+        Whether to execute the script in foreground mode or not.
+    log_config_dict : dict
+        Logging configuration dictionary.
+
+    Returns
+    -------
+    dict
+        Gunicorn configuration options.
+    """
+    # TODO: get values from the configuration
+    keyfile = '/var/ossec/api/configuration/ssl/server.key'
+    certfile = '/var/ossec/api/configuration/ssl/server.crt'
```

Comment:
These paths are going to be extracted from the configuration (which doesn't exist yet), since they are just the default path to a file and not a secret, I think we could leave them like this. They are hard-coded just to facilitate the API's testing